griptape/structures/structure.py

```diff
@@ -52,7 +57,8 @@ class Structure(ABC):
         ),
         kw_only=True,
     )
-    task_memory: Optional[TaskMemory] = field(
+    rag_engine: RagEngine = field(default=Factory(lambda self: self.default_rag_engine, takes_self=True), kw_only=True)
```

Comment:
Does this belong as a top level Structure field? Not every Structure will use RAG.

Reply:
I think it does since it's (optionally) reused in multiple places inside structures (e.g., `ToolMemory`, `TextArtifactStorage`, `RagTask` and probably more in the future). It's a very useful location to reference it from.